app/src/main/java/ch/epfl/sdp/ui/map/GoogleMapProvider.java

```diff
@@ -4,61 +4,69 @@
+    private boolean mHavePermission;
+    private Activity mActivity;
+    private Location mLocation;
+    private float mZoomLevel;
+    private GoogleMap mMap = null;
+
+    public GoogleMapProvider(Fragment fragment, MapView mapView){
+        this.mContext = fragment.getContext();
+        this.mActivity = fragment.getActivity();
+        mMarkerOptions = new ArrayList<>();
 
-    GoogleMapProvider(Context context,MapView mapView){
-        this.mMapView = mapView;
-        this.mContext = context;
+        // default current location
+        mLocation = new Location("Europe");
+        mLocation.setLatitude(46.520564);
+        mLocation.setLongitude(6.567827);
+        mZoomLevel = 4;
 
-        if (!mHavePermission) {
-            ActivityCompat.requestPermissions((Activity)context,
+        ActivityCompat.requestPermissions((Activity)mContext,
                     new String[]{Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION},
                     PERMISSION_LOCATION);
-        }
-        mHavePermission = ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
+
+        mHavePermission = ContextCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION)
                 == PackageManager.PERMISSION_GRANTED &&
-                ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION)
+                ContextCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_COARSE_LOCATION)
                         == PackageManager.PERMISSION_GRANTED;
 
+        if (mHavePermission){
+            LocationManager locationManager = (LocationManager) mActivity.getSystemService(mContext.LOCATION_SERVICE);
+            mLocation = locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+            mZoomLevel = 12;
+        }
         mapView.getMapAsync(this);
     }
 
     @Override
-    public void onMapReady(GoogleMap googlemap) {
-        mMap = googlemap;
-        mMap.getUiSettings().setMyLocationButtonEnabled(mLocationButtonEnabled&&mHavePermission);
-        mMap.setMyLocationEnabled(mLocationEnabled&&mHavePermission);
-        mMap.addMarker(mMarkerOptionsToBeAdded.iterator().next());
-        // for now display the french part of Switzerland on launch, to be modified
-        mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(46.520564, 6.567827), 9));
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+        setMapSettings(mMap, mMarkerOptions, mLocationEnabled&&mHavePermission);
+        googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(mLocation.getLatitude(), mLocation.getLongitude()), mZoomLevel));
     }
 
-    @Override
-    public void setMyLocationButtonEnabled(boolean enabled) {
-        mLocationButtonEnabled= enabled;
+    static public void setMapSettings(GoogleMap googleMap, List<MarkerOptions> markerOptionsList, boolean locationEnabled) {
```

Comment:
not a big fan of what's going on here

Reply:
c'est pour pouvoir tester une partie de ce qui est dans onMapReady